test/test_testing.py

```diff
@@ -2851,6 +2851,29 @@ def _on_collect_unit_status(self, event: ops.CollectStatusEvent):
         self.assertEqual(harness.model.app.status, ops.ActiveStatus('active app'))
         self.assertEqual(harness.model.unit.status, ops.ActiveStatus('active unit'))
 
+    def test_invalid_status_set(self):
```

Comment:
I don't think this test needs to go via collect-status, we can just set status directly. What about this:

```python
    def test_invalid_status_set(self):
        harness = ops.testing.Harness(ops.CharmBase)
        harness.set_leader(True)
        harness.begin()

        with self.assertRaises(ops.model.ModelError):
            harness.model.app.status = ops.UnknownStatus()
        with self.assertRaises(ops.model.ModelError):
            harness.model.app.status = ops.ErrorStatus()
        harness.model.app.status = ops.ActiveStatus()

        with self.assertRaises(ops.model.ModelError):
            harness.model.unit.status = ops.UnknownStatus()
        with self.assertRaises(ops.model.ModelError):
            harness.model.unit.status = ops.ErrorStatus()
        harness.model.unit.status = ops.ActiveStatus()
```

Reply:
Much direct, updated thank you!